src/libraries/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -85,31 +84,38 @@ public BitArray(byte[] bytes)
             m_array = new int[GetInt32ArrayLengthFromByteLength(bytes.Length)];
             m_length = bytes.Length * BitsPerByte;
 
-            uint totalCount = (uint)bytes.Length / 4;
-
-            ReadOnlySpan<byte> byteSpan = bytes;
-            for (int i = 0; i < totalCount; i++)
+            if (BitConverter.IsLittleEndian)
             {
-                m_array[i] = BinaryPrimitives.ReadInt32LittleEndian(byteSpan);
-                byteSpan = byteSpan.Slice(4);
+                bytes.CopyTo(MemoryMarshal.AsBytes(m_array.AsSpan()));
             }
+            else
+            {
+                int totalCount = bytes.Length / 4;
 
-            Debug.Assert(byteSpan.Length >= 0 && byteSpan.Length < 4);
+                ReadOnlySpan<byte> byteSpan = bytes;
+                for (int i = 0; i < totalCount; i++)
+                {
+                    m_array[i] = BinaryPrimitives.ReadInt32LittleEndian(byteSpan);
+                    byteSpan = byteSpan.Slice(4);
+                }
 
-            int last = 0;
-            switch (byteSpan.Length)
-            {
-                case 3:
-                    last = byteSpan[2] << 16;
-                    goto case 2;
-                // fall through
-                case 2:
-                    last |= byteSpan[1] << 8;
-                    goto case 1;
-                // fall through
-                case 1:
-                    m_array[totalCount] = last | byteSpan[0];
-                    break;
+                Debug.Assert(byteSpan.Length >= 0 && byteSpan.Length < 4);
+
+                int last = 0;
+                switch (byteSpan.Length)
+                {
+                    case 3:
+                        last = byteSpan[2] << 16;
+                        goto case 2;
+                    // fall through
```

Comment:
I know these are being caried over from the previous, but the comments are superfluous. The `goto case #` makes it explicit where the control flow goes and so it isn't "technically" fallthrough and more-so just adds noise.

Reply:
If I am reading it right, this block (lines 87 - 118) can be

```c#
bytes.CopyTo(MemoryMarshal.AsBytes(m_array.AsSpan()));

if (!BitConverter.IsLittleEndian)
{
    BinaryPrimitives.ReverseEndianness<int>(m_array, m_array);
}
```